mycpp/marksweep_heap.h

```diff
@@ -75,6 +76,13 @@ class RootSet {
     stack_[num_frames_ - 1].push_back(root);
   }
 
+  void AddGlobalRoot(Obj* root) {
```

Comment:
This is the right idea, but we should have tests for the new API.  I think it can go in marksweep_heap_test.cc:

1. it could just make something that assigns a GC object to a global
2. And then it should crash under `gcevery mode`
3. and then add AddGlobalRoot() , which should fix the crash I believe